scripts/check-ts.js

```diff
@@ -0,0 +1,82 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+const childProcess = require("child_process");
+
+const event = process.env.GITHUB_EVENT;
+
+const body = JSON.parse(event);
+const baseSHA = body.pull_request.base.sha;
+const headSHA = body.pull_request.head.sha;
+
+const gitLogFilesCommand = `git diff --name-only ${baseSHA}...${headSHA}`;
+const gitLogFilesResult = childProcess.execSync(gitLogFilesCommand, {
+  encoding: "utf8",
+});
+const filePaths = gitLogFilesResult.split(/\r?\n/);
+
+console.log("The following files will be reviewed:");
+
+const linesWithReferencesToDeprecatedCSS = [];
+const linesWithMissingTypes = [];
+const linesWithArrayType = [];
+
+for (let index = 0; index < filePaths.length; index++) {
+  const filePath = filePaths[index];
+  console.log(filePath);
+
+  if (filePath.endsWith(".ts") || filePath.endsWith(".tsx")) {
+    const gitLogCommand = `git diff ${baseSHA}...${headSHA} -- ${filePath}`;
+    const result = childProcess.execSync(gitLogCommand, { encoding: "utf8" });
+    const lines = result.split(/\r?\n/);
+
+    for (let index = 0; index < lines.length; index++) {
+      const line = lines[index];
+
+      if (line.startsWith("+")) {
+        if (line.includes("var(--")) {
+          linesWithReferencesToDeprecatedCSS.push(line);
+        } else if (
+          line.includes("useState(") ||
+          line.includes("useQuery(") ||
+          line.includes("useMutation(") ||
+          line.includes("useSubscription(") ||
+          line.includes("useContext(") ||
+          line.includes("useRef(") ||
+          line.includes("useMachine(")
+        ) {
+          linesWithMissingTypes.push(line);
+        } else if (line.includes(": Array<")) {
+          linesWithArrayType.push(line);
```

Comment:
Maybe, also write the file path along with the line could be better, especially for significant PR

Reply:
`push(`${filePath}:${line}`) ?